lib/ViewModels/WorkbenchItemControls.ts

```diff
@@ -0,0 +1,27 @@
+import remove from "lodash-es/remove";
+import { action } from "mobx";
+import { BaseModel } from "../Models/Definition/Model";
+import { SelectableDimension } from "../Models/SelectableDimensions/SelectableDimensions";
+import ViewState from "../ReactViewModels/ViewState";
+
+type WorkbenchItemControlsGenerator = (
+  item: BaseModel
+) => SelectableDimension | undefined;
+
+type WorkbenchItemControlDisposer = () => void;
+
+export const addControl = action(
```

Comment:
Need jsdoc for this, to help intellisense be clear on what the generator function should return. When testing it was not obvious that my generator Fn had to return something like ```{
          type: "button",
          value: "Test Button",
          setDimensionValue: () => console.log('xxxxxxxx'),
          buttonStyle: "secondary"
        }
        ```

Reply:
In fact, I find `setDimensionValue` to be not a clear description for what is essentially the function that gets called onClick for the button thats added... I suppose this is defined in SelectableDimensions and that is a core Terria concept. Perhaps it is just because Im not familiar with the Selectable Dimensions concept yet, but is there any way we could ovveride this, and when people use `WorkbenchItemControls.addControl` in their plugins they specify `onClick` function instead or something that is less specific to Terria in house nomenclature?

Reply:
I also found that it was not immediately obvious that `addControl` would add a control for every item added to the workbench (subject to conditions of course). Perhaps this funciton could be renamed `addControlForItem` to make that clear?